src/Features/Core/Portable/NavigateTo/WrappedNavigateToSearchService.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.NavigateTo
+{
+    internal partial class NavigateToSearcher
+    {
+        private class WrappedNavigateToSearchService : INavigateToSearchService
+        {
+#pragma warning disable CS0618 // Type or member is obsolete
+            private readonly INavigateToSeINavigateToSearchService_RemoveInterfaceAboveAndRenameThisAfterInternalsVisibleToUsersUpdatearchService _legacySearchService;
```

Comment:
I've never seen this pattern of annoyingly-long-names-that-make-it-really-obvious-something-needs-changing before Roslyn, and I really love it.